src/Components/animation/Animation.js

```diff
@@ -137,377 +69,265 @@ class Animation extends React.Component {
+        stepId = Math.max(0, Math.round(stepId));
+
+        var sequence = [];
+
+        /* Move instantly to the previous step */
+        sequence.push(
+          Animated.timing(this.currentStepAV, {
+            toValue: Math.max(0, stepId - 1),
+            duration: 0,
+            easing: Easing.linear,
+            key: 0,
+          }),
+        );
 
-          displayedElementStepAnimation.push(anim);
+        /* If we do not move to the first step, animate the move from the previous step to the current step */
+        if (stepId > 0) {
+          sequence.push(
+            Animated.timing(this.currentStepAV, {
+              toValue: stepId,
+              duration: this.state.stepLength,
+              easing: Easing.linear,
+              key: 1,
+            }),
+          );
         }
 
-        stepAnimation.push(Animated.sequence(displayedElementStepAnimation));
-      }
-    }
-
-    return stepAnimation;
+        Animated.sequence(sequence).start(() => this.setState({ animationPlaying: false, currentStep: stepId }));
+      },
+    );
   };
 
-  _display(item) {
-    if (item !== undefined && item !== null) return item.render();
-    else return <View />;
-  }
-
   render() {
-    this.cutsArray = [];
+    //<MaterialCommunityIcons name="chevron-double-left" color={theme.COLOR_PRIMARY} size={26} />
 
     return (
       <View style={[styles.mainContainer, { height: this.animationHeight }, { width: this.animationWidth }]}>
-        {this.props.editable && this.cuts !== undefined && this.cuts !== null ? (
-          <DisplayedCuts step={this.state.currentStep} drillCuts={this.cuts} />
+        {this.props.editable && !this.state.animationPlaying ? (
+          <DisplayedCuts
+            step={this.state.currentStep}
+            positionPercentToPixel={this._positionPercentToPixel}
+            animation={this.props.animation}
+            onMoveEnd={this.props.onCutMove}
+          />
         ) : (
           undefined
         )}
-        {this.state.displayedElements === undefined || this.state.displayedElements === null ? (
-          <View />
-        ) : (
-          this.state.displayedElements.map(this._display)
-        )}
-        <View style={styles.controls}>
-          <TouchableOpacity style={styles.controlBtn} onPress={() => this._previousStep()}>
+
+        {this.state.displayedElements?.map(item => {
+          return (
+            <DisplayedElement
+              id={item.id}
+              number={item.number}
+              key={item.key}
+              eId={item.eId}
+              movable={item.movable}
+              onMoveEnd={item.onMoveEnd}
+              animationWidth={this.animationWidth}
+              animationHeight={this.animationHeight}
+              animation={this.props.animation}
+              currentStepAV={this.currentStepAV}
+              positionPercentToPixel={this._positionPercentToPixel}
+            />
+          );
+        })}
+
+        {/* {this.props.editable ? (
```

Comment:
Maybe remove this code if we don't need it anymore?

Reply:
Done